extensions/src/main/java/org/pentaho/platform/web/http/api/resources/Log4jResource.java

```diff
@@ -0,0 +1,103 @@
+/*!
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2017 Pentaho Corporation..  All rights reserved.
+ */
+
+package org.pentaho.platform.web.http.api.resources;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Level;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.apache.log4j.helpers.Loader;
+import org.apache.log4j.xml.DOMConfigurator;
+import org.codehaus.enunciate.jaxrs.ResponseCode;
+import org.codehaus.enunciate.jaxrs.StatusCodes;
+
+import javax.ws.rs.FormParam;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Enumeration;
+
+
+/**
+ * Created by on 6/23/2017.
+ */
+@Path( "/logconfig" )
+public class Log4jResource {
+
+  private static final Logger LOGGER = Logger.getLogger( Log4jResource.class );
+  private static final String CONFIG = "log4j.xml";
+
+  @PUT
+  @Path ( "/reset" )
+  @StatusCodes( {
+    @ResponseCode( code = 200, condition = "Successfully reset log level." )
+    } )
+  @Produces( { MediaType.TEXT_PLAIN } )
+  public Response resetLogLevel() throws Exception {
+    LOGGER.setLevel( Level.INFO );
```

Comment:
Can we assume that by "reset" we always want to set the log level to "info"? It might make sense to have the "default level" be configurable, so the user can control what "resetting" really means